upcloud/error_utils.go

```diff
@@ -0,0 +1,87 @@
+package upcloud
+
+import (
+	"net/http"
+	"strings"
+)
+
+// IsNotFoundError checks if provided error is API Not Found error
+func IsNotFoundError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	if uError, ok := err.(*Error); ok {
+		return uError.Status == http.StatusNotFound
+	}
+
+	if uProblem, ok := err.(*Problem); ok {
+		return uProblem.Status == http.StatusNotFound
+	}
```

Comment:
IMHO we should have only one error type which would make this function helper cleaner or even unnecessary.

Reply:
I agree, but doing it at this point would probably break backwards compatibility. Not sure how much people used to rely on `Error` and `Problem` types tho.